pdblp/tests/test_pdblp.py

```diff
@@ -164,3 +164,19 @@ def test_hist_ref_one_ticker_one_field_pivoted_non_numeric(self):
         )
         df_expect.index.names = ["date"]
         assert_frame_equal(df, df_expect)
+
+    def test_beqs_current(self):
+        df = self.con.beqs('Insider Buyers', 'GLOBAL', 'Popular')
+        assert len(df) > 0
+
+    def test_beqs_asof(self):
+        asof_date = datetime.datetime(2016, 5, 4)
+        df = self.con.beqs('Insider Buyers', 'GLOBAL', 'Popular', asof_date=asof_date)
+        assert len(df) == 1145
+
+    def test_beqs_hist(self):
```

Comment:
Test is failing for me since I am getting a DataFrame of length 488. Bloombergs exact data returned is not always reliable, a test which just checks the known properties of the returned DataFrame would be better (e.g. a DataFrame with specific columns)

Reply:
Hello Matthew,

I have read your codes and also the codes from blpapi. I found it is impossible to get the data of backdates. Actually I want to realize the lookup function that use the market cap rank in 2019 to select stock. But I cannot find any api methods to do that. I have found yours codes is working, but it is now not accurate?

Beth